packages/metrics/tests/unit/Metrics.test.ts

```diff
@@ -2343,6 +2366,54 @@ describe('Class: Metrics', () => {
     });
   });
 
+  describe('Method: isDisabled', () => {
+    it('should be enabled by default', () => {
+      const metrics: Metrics = new Metrics({ namespace: TEST_NAMESPACE });
+
+      // Act & Assess
+      // biome-ignore lint/complexity/useLiteralKeys: accessing protected method
+      expect(metrics['isDisabled']()).toBe(false);
```

Comment:
Since `isDisabled` is `protected`, TypeScript is reporting an error if accessed via dot notation.
The bracket notation is allowed by TS as [intentional escape hatch](https://github.com/microsoft/TypeScript/issues/19335). 

I could also refactor this tests to use `publishStoredMetrics()` instead. IMO prefer the cleanliness of the tests, but I'm open to changing it.

Reply:
Probably obsolete if we change `getMetricsDisabled`

Reply:
I think it's fine to leave it as you wrote it for now, and is consistent with other tests.

I am also working on another PR that refactors the unit tests for this module to address this (#2821) and remove this type of pattern in favor of the other one you're suggesting. Overall I think observing the actual side effect - in this case the metrics not being emitted - is better and makes the tests less brittle and maintainable over time.

With this type of test that reaches into the implementation, if we for example rename the `isDisabled` property or refactor the internal implementation, the test will fail and also need refactoring while the feature hasn't functionally changed.

If instead we observe the behavior, as long as metrics are not published the refactor was successful.

---

For now however leave it as-is as it'll make the rebasing easier to follow, since using the other pattern would require more changes. If you're curious about how the refactoring looks like, check the PRs in a couple hrs. I was waiting to publish that branch to save you from some merge conflicts.

Reply:
Okay, let me know if need to change something after your PR is merged! 👍